spec/app/domain/authentication/authn-gcp/extract_resource_restrictions_spec.rb

```diff
@@ -0,0 +1,440 @@
+    allow(mocked_role_class_return_annotations_list).to receive(:annotations)
+                                                          .and_return(annotations_list)
+
+    allow(mocked_resource_class_return_annotations_nil).to receive(:[])
+                                                             .with(any_args)
+                                                             .and_return(mocked_role_class_return_annotations_nil)
+    allow(mocked_role_class_return_annotations_nil).to receive(:annotations)
+                                                         .and_return(nil)
+
+    allow(mocked_resource_class_return_annotations_empty_list).to receive(:[])
+                                                                    .with(any_args)
+                                                                    .and_return(mocked_role_class_return_annotations_empty_list)
+    allow(mocked_role_class_return_annotations_empty_list).to receive(:annotations)
+                                                                .and_return([])
+
+    allow(mocked_resource_class_return_annotations_list_with_duplications).to receive(:[])
+                                                                                .with(any_args)
+                                                                                .and_return(mocked_role_class_return_annotations_list_with_duplication)
+    allow(mocked_role_class_return_annotations_list_with_duplication).to receive(:annotations)
+                                                                           .and_return(annotations_list_with_duplications)
+
+
+  end
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "An invalid input parameters to extract resource restrictions" do
+    context "when account not exists" do
+      subject do
+        Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+          resource_class: mocked_resource_class_return_nil
+        ).call(
+          account:           invalid_account,
+          username:          valid_host,
+          extraction_prefix: valid_prefix
+        )
+      end
+
+      it "raises an error" do
+        expect {subject}.to raise_error(Errors::Authentication::Security::RoleNotFound)
+      end
+
+      context "when host not exists" do
+        subject do
+          Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+            resource_class: mocked_resource_class_return_nil
+          ).call(
+            account:           valid_account,
+            username:          invalid_host,
+            extraction_prefix: valid_prefix
+          )
+        end
+
+        it "raises an error" do
+          expect {subject}.to raise_error(Errors::Authentication::Security::RoleNotFound)
```

Comment:
I think better validation will be `InvalidHostId` error

Reply:
why? the user gave us a hostname and it is not found in the db

Reply:
username give invalid_host , i think the error InvalidHostId is the relevant error for this. RoleNotFound its too general imo.

Reply:
that's just a naming decision. @nessiLahav could have just called it `non_existing_host`. in any case, what we verify here is that the host doesn't exist.